code/game/objects/structures/aliens.dm

```diff
@@ -112,38 +247,93 @@
 	desc = "A thick resin surface covers the floor."
 	anchored = TRUE
 	density = FALSE
-	layer = ABOVE_OPEN_TURF_LAYER
 	plane = FLOOR_PLANE
-	icon = 'icons/obj/smooth_structures/alien/weeds1.dmi'
-	icon_state = "weeds1"
-	base_icon_state = "weeds1"
+	icon = 'icons/obj/smooth_structures/alien/weeds.dmi'
+	icon_state = "weeds"
+	base_icon_state = "weeds"
 	max_integrity = 15
+	layer = WIRE_TERMINAL_LAYER
 	smoothing_flags = SMOOTH_BITMASK
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WEEDS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WEEDS, SMOOTH_GROUP_WALLS)
-	transform = matrix(1, 0, -4, 0, 1, -4)
 	var/obj/structure/alien/weeds/node/linked_node = null
+	var/obj/structure/alien/wallweed/wall_weed
 	var/static/list/weedImageCache
+	var/check_counter
 
-
-/obj/structure/alien/weeds/New(pos, node)
+/obj/structure/alien/weeds/Initialize(mapload, node)
 	..()
 	linked_node = node
 	if(isspaceturf(loc))
 		qdel(src)
 		return
-	if(icon_state == "weeds")
-		icon_state = pick("weeds", "weeds1", "weeds2")
-	spawn(rand(150, 200))
-		if(src)
-			Life()
+	check_surroundings()
+	START_PROCESSING(SSobj, src)
 
 /obj/structure/alien/weeds/Destroy()
+	STOP_PROCESSING(SSobj, src)
 	QUEUE_SMOOTH_NEIGHBORS(src)
+	playsound(loc, pick('sound/effects/alien_resin_break2.ogg','sound/effects/alien_resin_break1.ogg'), 50, FALSE)
 	linked_node = null
+	clear_wall_weed()
 	return ..()
 
-/obj/structure/alien/weeds/proc/Life()
+/obj/structure/alien/weeds/attack_alien(mob/living/carbon/alien/humanoid/user)
+	if(user.a_intent != INTENT_HARM)
+		return
+	return ..()
+
+/obj/structure/alien/weeds/process()
+    check_counter++
+    if(check_counter >= 5)
+        spread()
+        check_counter = 0
+
+/obj/structure/alien/weeds/proc/clear_wall_weed()
+	if(wall_weed)
+		qdel(wall_weed)
+		wall_weed = null
+
+/obj/structure/alien/weeds/proc/check_surroundings()
+	var/turf/T = get_turf(src)
+	var/list/nearby_dense_turfs = T.AdjacentTurfs(cardinal_only = FALSE, dense_only = TRUE)
+	if(!length(nearby_dense_turfs)) // There is no dense turfs around it
+		clear_wall_weed()
+		return
+
+	var/list/wall_dirs = list()
+	for(var/turf/W in nearby_dense_turfs)
+		if(iswallturf(W))
+			wall_dirs.Add(get_dir(W, T))
+	if(!length(wall_dirs)) // There is no walls around it
+		clear_wall_weed()
+		return
+
+	var/list/nearby_open_turfs = T.AdjacentTurfs(open_only = TRUE, cardinal_only = TRUE)
+
+	for(var/turf/W in nearby_open_turfs)
+		if(locate(/obj/structure/alien/weeds, W))
+			var/dirs = get_dir(W, T)
+			switch(dirs)
+				if(NORTH)
+					wall_dirs.Remove(NORTHEAST, NORTHWEST)
+				if(SOUTH)
+					wall_dirs.Remove(SOUTHEAST, SOUTHWEST)
+				if(EAST)
+					wall_dirs.Remove(NORTHEAST, SOUTHEAST)
+				if(WEST)
+					wall_dirs.Remove(NORTHWEST, SOUTHWEST)
```

Comment:
can you document what this section of code does

Reply:
Donezo